File: dodam-system-domain/dodam-domain-rds/src/main/java/b1nd/dodam/domain/rds/nightstudy/repository/NightStudyBanRepository.java
package b1nd.dodam.domain.rds.nightstudy.repository;

import b1nd.dodam.domain.rds.member.entity.Student;
import b1nd.dodam.domain.rds.nightstudy.entity.NightStudyBan;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDate;
import java.util.List;
import java.util.Optional;

public interface NightStudyBanRepository extends JpaRepository<NightStudyBan, Long> {

    Optional<NightStudyBan> findByStudent(Student student);
    
    Optional<NightStudyBan> findByStudentAndEndedGreaterThanEqual(Student student, LocalDate today);

    @EntityGraph(attributePaths = {"student", "student.member"})
    List<NightStudyBan> findByEndedGreaterThanEqual(LocalDate today);

    List<NightStudyBan> findByStudentInAndEndedGreaterThanEqual(List<Student> students, LocalDate today);

    @Override
    @EntityGraph(attributePaths = {"student", "student.member"})
    List<NightStudyBan> findAll();
}

File: services/service-night-study/src/main/kotlin/com/b1nd/dodamdodam/nightstudy/domain/ban/repository/NightStudyBanRepository.kt
package com.b1nd.dodamdodam.nightstudy.domain.ban.repository

import com.b1nd.dodamdodam.nightstudy.domain.ban.entity.NightStudyBanEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDate
import java.util.UUID

interface NightStudyBanRepository : JpaRepository<NightStudyBanEntity, Long> {

    fun findAllByUserId(userId: UUID): List<NightStudyBanEntity>

    fun findAllByUserIdAndEndedAtGreaterThanEqual(userId: UUID, date: LocalDate): List<NightStudyBanEntity>

    fun deleteAllByUserId(userId: UUID)
}
